File: backend/app/schemas/imports.py
from __future__ import annotations

from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.models.main_projects import MainProjectStatus


class ProjectImportRowErrorRead(BaseModel):
    row_number: int
    field: str
    message: str
    value: str | None = None


class ProjectImportCreatedProjectRead(BaseModel):
    id: UUID
    project_no: str
    name: str
    dept_id: UUID
    status: MainProjectStatus

    model_config = ConfigDict(from_attributes=True)


class ProjectImportResultRead(BaseModel):
    batch_no: str
    total_rows: int
    success_count: int
    failure_count: int
    duration_ms: int
    errors: list[ProjectImportRowErrorRead]
    created_projects: list[ProjectImportCreatedProjectRead]

File: frontend/src/types/imports.ts
import type { ProjectStatus } from '@/types/projects';

export interface ProjectImportRowErrorRead {
  field: string;
  message: string;
  row_number: number;
  value: string | null;
}

export interface ProjectImportCreatedProjectRead {
  dept_id: string;
  id: string;
  name: string;
  project_no: string;
  status: ProjectStatus;
}

export interface ProjectImportResultRead {
  batch_no: string;
  created_projects: ProjectImportCreatedProjectRead[];
  duration_ms: number;
  errors: ProjectImportRowErrorRead[];
  failure_count: number;
  success_count: number;
  total_rows: number;
}
